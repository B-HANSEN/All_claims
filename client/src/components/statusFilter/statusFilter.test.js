import { render, screen, fireEvent } from '@testing-library/react';
import ShallowRenderer from 'react-test-renderer/shallow';
import React from 'react';
import StatusFilter from '.';

describe('Status filter tests', () => {
	it('should match snapshot', () => {
		const renderer = new ShallowRenderer();
		const result = renderer.render(<StatusFilter />);
		expect(result).toMatchSnapshot();
	});

	it('should update local state with new filter value', () => {
		const handleFilter = jest.fn();

		render(<StatusFilter filter={'ShowAll'} setFilter={handleFilter} />);
		expect(screen.getByText('Processed')).toBeInTheDocument();

		fireEvent.change(screen.getByTestId('select'), {
			target: { value: 'Approved' },
		});
		expect(handleFilter).toHaveBeenCalledTimes(1);
	});
});
