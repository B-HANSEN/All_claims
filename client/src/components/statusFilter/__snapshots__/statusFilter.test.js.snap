// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Status filter tests should match snapshot 1`] = `
<div
  className="StatusFilter"
>
  <p>
    Filter by claim status
  </p>
  <FormSelect
    aria-label="Default select example"
    className="StatusFilter__Select"
    data-testid="select"
    id="status"
    onChange={[Function]}
    size="lg"
  >
    <option
      value="ShowAll"
    >
      Show All
    </option>
    <option
      value="Submitted"
    >
      Submitted
    </option>
    <option
      value="Approved"
    >
      Approved
    </option>
    <option
      value="Processed"
    >
      Processed
    </option>
    <option
      value="Completed"
    >
      Completed
    </option>
    <option
      value="Rejected"
    >
      Rejected
    </option>
  </FormSelect>
</div>
`;
